caffe2/operators/spatial_batch_norm_op.h

```diff
@@ -90,35 +90,21 @@ class SpatialBNOp : public Operator<Context> {
           alpha_data,
           beta_data);
     } else {
-      auto* saved_mean = Output(SAVED_MEAN);
-      auto* saved_rstd = Output(SAVED_INV_STD);
-      if (num_batches_ == 1) {
-        saved_mean->Resize(C);
-        saved_rstd->Resize(C);
-      } else {
-        const auto& batch_mean_sum = Input(BATCH_MEAN_SUM);
-        const auto& batch_var_sum = Input(BATCH_VAR_SUM);
-        if (saved_mean != &batch_mean_sum) {
-          saved_mean->Resize(C);
-        }
-        if (saved_rstd != &batch_var_sum) {
-          saved_rstd->Resize(C);
-        }
-      }
+      auto* saved_mean = Output(SAVED_MEAN, {C}, at::dtype<T>());
+      auto* saved_rstd = Output(SAVED_INV_STD, {C}, at::dtype<T>());
       T* saved_mean_data = saved_mean->template mutable_data<T>();
       T* saved_rstd_data = saved_rstd->template mutable_data<T>();
-      auto* running_mean = Output(RUNNING_MEAN);
-      auto* running_var = Output(RUNNING_VAR);
-      if (running_mean->numel() != C) {
-        running_mean->Resize(C);
-        math::Set<T, Context>(
-            C, T(0), running_mean->template mutable_data<T>(), &context_);
-      }
-      if (running_var->numel() != C) {
-        running_var->Resize(C);
-        math::Set<T, Context>(
-            C, T(0), running_var->template mutable_data<T>(), &context_);
-      }
+
+      // Enforce Alias
+      CAFFE_ENFORCE(
```

Comment:
I don't really understand the code transformation here. Please double check this enforces the right thing.
This assertion ensure that you're in the branch that would **not** have called the resize in line 102 and 105. Is this what you meant? Because your `Output` is doing a resize.

Reply:
I'm not sure why this original code is written in this way, but we have a enforce in schema that requires them to be alias, we can probably remove the enforce here as well: https://github.com/pytorch/pytorch/blob/master/caffe2/operators/spatial_batch_norm_op.cc#L33

Reply:
Better to enforce it here just in case :)